src/storage/v2/vertex_accessor.cpp

```diff
@@ -301,7 +301,13 @@ Result<bool> VertexAccessor::InitProperties(const std::map<storage::PropertyId,
   bool result{false};
   utils::AtomicMemoryBlock atomic_memory_block{
       [&result, &properties, storage = storage_, transaction = transaction_, vertex = vertex_]() {
-        if (!vertex->properties.InitProperties(properties)) {
+        if (std::ranges::any_of(vertex->labels,
```

Comment:
We should probably add benchmark, something like:
* 5-10 props
* 1/2/3 labels per node
* 0 | 5 text search indexes
How fast insertion of a new vertex is?

Reply:
We now have a workload PR in the private repo for this